src/auth/auth.service.test.ts

```diff
@@ -0,0 +1,42 @@
+import { ConfigService } from '@nestjs/config';
+import { PassportModule } from '@nestjs/passport';
+import { Test, TestingModule } from '@nestjs/testing';
+import Chance from 'chance';
+
+import { AuthService } from './auth.service';
+import { ApiKeyStrategy } from './strategy/apikey.strategy';
+
+describe('AuthService', () => {
+  let spyService: AuthService;
+  const chance = new Chance();
+
+  beforeAll(async () => {
+    const app: TestingModule = await Test.createTestingModule({
+      imports: [PassportModule],
+      providers: [
+        AuthService,
+        ApiKeyStrategy,
+        ConfigService,
+        {
+          provide: AuthService,
+          useValue: {
+            validateApiKey: jest.fn().mockResolvedValue(true),
+          },
+        },
+      ],
+    }).compile();
+
+    spyService = app.get<AuthService>(AuthService);
+  });
+
+  it('should be defined', () => {
+    expect(spyService).toBeDefined();
+  });
+
+  it('should call `validateApiKey` with expected param', () => {
+    const key = chance.word();
+    spyService.validateApiKey(key);
+
+    expect(spyService.validateApiKey).toHaveBeenCalledWith(key);
+  });
```

Comment:
I don't think this test is quite right: line 40 will always pass because of line 38, we're not testing any code within the real `AuthService`

I expected instead that the test would:

1. mock `ConfigService` to return a particular value for `'app.apiKey'`
2. call `authService.validateApiKey` with a real auth service
3. check that it returns true if the input matches the mocked value from step 1, or false if it does not

Reply:
this has been updated